backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/radiobutton/BpkRadioButton.kt

```diff
@@ -83,11 +84,42 @@ fun BpkRadioButton(
     modifier = rowModifier,
     verticalAlignment = Alignment.CenterVertically,
   ) {
-    BpkRadioButton(selected = selected, onClick = onClick, enabled = enabled, interactionSource = interactionSource)
-    BpkText(
-      text = text,
-      style = BpkTheme.typography.footnote,
-      color = if (enabled) BpkTheme.colors.textPrimary else dynamicColorOf(BpkColor.SkyGrayTint04, BpkColor.BlackTint06),
+    BpkRadioButtonImpl(selected = selected, onClick = onClick, enabled = enabled, interactionSource = interactionSource)
+    val contentColor =
+      if (enabled) BpkTheme.colors.textPrimary else dynamicColorOf(BpkColor.SkyGrayTint04, BpkColor.BlackTint06)
+    CompositionLocalProvider(
+      LocalContentColor provides contentColor,
+      LocalContentAlpha provides 1f,
+      LocalTextStyle provides BpkTheme.typography.footnote,
+    ) {
+      content(selected)
+    }
+  }
+}
+
+@OptIn(ExperimentalMaterialApi::class, ExperimentalComposeUiApi::class)
+@Composable
+private fun BpkRadioButtonImpl(
+  selected: Boolean,
+  onClick: (() -> Unit)?,
+  modifier: Modifier = Modifier,
+  enabled: Boolean = true,
+  interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+) {
+  // our design system isn't designed with the minimum touch target in mind at the moment.
+  // Disable the enforcement to avoid the extra padding
+  CompositionLocalProvider(LocalMinimumTouchTargetEnforcement provides false) {
+    RadioButton(
+      selected = selected,
+      onClick = onClick,
+      modifier = modifier.semantics { invisibleToUser() },
```

Comment:
that's the one change to this function - make it invisible to the user - this makes it not focusable in accessibility mode as the whole row has the radiobutton role & handles the functionality (and provides the context around it as well)

Reply:
I'd probably inline it and not make it as a function.

Reply:
i'd rather keep it as a separate function as it is cleaner. inlining has minimal benefit for this case based on the IDE hint, so don't think that's necessary